eng/pipelines/templates/stages/archetype-net-release.yml

```diff
@@ -276,3 +270,34 @@ stages:
             command: push
             packagesToPush: '$(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.name}}/*.nupkg;!$(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.name}}/*.symbols.nupkg'
             publishVstsFeed: $(DevOpsFeedID)
+
+    - job: PublishDocsToNightlyBranch
+      dependsOn: PublishPackages
+      condition: or(eq(variables['SetDevVersion'], 'true'), and(eq(variables['SetDevVersion'], ''), eq(variables['Build.Reason'],'Schedule'), eq(variables['System.TeamProject'], 'internal')))
```

Comment:
Why do you have the SetDevVersion="" as part of this condition? We should probably make this condition match the publish to dev feed job.

Reply:
This is a copy/paste from the JS job. https://github.com/Azure/azure-sdk-for-js/blob/main/eng/pipelines/templates/stages/archetype-js-release.yml#L278

The JS job's criteria for `PublishPackages` and `PublishDocsToNightlyBranch` are the same and that criteria is different from the .NET criteria for similar jobs. 

I'm fine with using the .NET criteria here. I'm not sure why these are different. Perhaps that should be cleaned up later. https://github.com/Azure/azure-sdk-tools/issues/2037

Reply:
Yeah as far as I can remember the we don't need the SetDevVersion="" part of this condition but it probably doesn't harm anything in our usages either.